### ISSUE TITLE
Version 0.9.49 halts download even while using internal storage (Android)

### ISSUE THREAD

**Issue by KillerJoeBR** [scakemyer/plugin.video.quasar#620]
New version actually stopped working my downloads while using internal storage. I tried using the sdcard instead and the same thing happens. I couldn't download anything. It starts fine but after a few seconds it would stop downloading. 

I downgraded to Quasar 0.9.48 and it works ok now. (same download path)

Later i will submit the Log.

**Comment by scakemyer** [comment]
Thanks for the feedback, and yes will definitely need the log.

**Comment by KillerJoeBR** [comment]
Here we go:

http://pastebin.com/fDuU4Tug

https://www.dropbox.com/s/rqdpcxsvk45bnzl/spmc.log?dl=0

**Comment by scakemyer** [comment]
So glad I enabled libtorrent error alerts, at least I know it either has to do with libtorrent itself or the golang bindings, although I'm guessing it's the former since it manages all of that.
```
torrent_error_alert: ... ERROR: Invalid argument
file_error_alert: ... open (/storage/emulated/0/Download/.../...) error: Invalid argument
```
Looking into it and will update if I find a relevant issue over there.

**Comment by scakemyer** [comment]
Can't find anything specific to those errors yet.. Do files show up in your download folder? There's already a few fixes related to file_storage since libtorrent 1.1.1, if you're willing to try a build based on the release candidate branch, we'd know if it's already fixed.

**Comment by scakemyer** [comment]
Noticed it loaded fastresume files, can you try removing them before starting your downloads again? Something tells me the format has changed and adding those might be tripping up libtorrent.

**Comment by KillerJoeBR** [comment]
I would like to try the libtorrent 1.1.1 , just point me in the right direction please. And how do i remove the fastresume files?

**Comment by scakemyer** [comment]
They're located in your Quasar download folder under Torrents. You can use the file manager in Kodi to do so, just make sure you remove active ones from Quasar first.

**Comment by scakemyer** [comment]
Once you've done that, let's start by upgrading to 0.9.49 and trying the same torrent with all resumefiles removed. If we still get the same errors I'll make a build from RC_1_1.

**Comment by KillerJoeBR** [comment]
I deleted the torrents folder and tried it again and still the same thing happened. I changed my Quasar settings to keep the download and i verified that the video file was created in my download path. So it works for a short time then halts.

**Comment by scakemyer** [comment]
Ok so that's a good thing we just singled out. Do you have logs for this run? I'll start the custom build right away.

**Comment by KillerJoeBR** [comment]
I will get a new log. Just wanted to say that i did more tests and the problem seems to only happen when the download file is bigger than 2gb or something like that. When the file was about 1gb it works fine (the video plays), but if i choose one about 3gb then the download halts after a few seconds during buffering.

**Comment by scakemyer** [comment]
Really sounds like a libtorrent thing. Here's your custom build using RC_1_1: https://offshoregit.com/scakemyer/plugin.video.quasar-0.9.49.android_arm.zip
Let me know how that goes.

**Comment by KillerJoeBR** [comment]
https://www.dropbox.com/s/rqdpcxsvk45bnzl/spmc.log?dl=0

Ok so first i tried 1 movie at 1.3gb and it worked. Then i tried another one at 3.4gb and it crashed (don't know if it was related or just coincidence) and then again another 3.4gb and the download after a few seconds halted.

**Comment by KillerJoeBR** [comment]
Well, tried the new Quasar version and it seems to be working fine again while using internal storage. Tried a couple of movies and they played fine.

Then i tried using a sdcard as download path and the download didn't actually stop but after a few seconds the download speed gets really low (like 50Kb/s) but it kept going. Maybe my sdcard is not good enough, but it's unusable. But at least internal storage is working again. I will test some more! 

Thanks!

**Comment by KillerJoeBR** [comment]
The last one improved the download speed while using SD card but when the buffering completes and the movie should start playing it hangs and then give me an error message.  But like I said before it could be my SD card is not good enough to write and read at the same time.  Would be nice if someone else could report it.

**Comment by scakemyer** [comment]
Yeah we need more feedback on this... as mentioned in the other issue I got it working pretty well with a NAS setup, so your SD card really must be junk ;) What's the error message you're getting?

**Comment by KillerJoeBR** [comment]
Ok i did some more tests and now i think the issue is not related to my sdcard performance (at least not entirely). I tried using my usbdrive which i used in the past and it did work fine before. Anyway now the same thing happens using it instead of the sdcard. The download goes ok until the buffering is completed and then it halts when the movie should start playing and then after a while it gives an error message (it's in portuguese but it's a general message about it has failed to reproduce the video, please check the log)

After this, the entire system slows down. If i try to open the storage manager it hangs for about 30 seconds and after a while Quasar crashes and starts again and the system is ok again (speed wise). Even if i exit kodi and try another android app , it opens slowly during this, and after quasar crashes it's ok again)

Finally here is the log: https://www.dropbox.com/s/rqdpcxsvk45bnzl/spmc.log?dl=0

**Comment by scakemyer** [comment]
Sadly there's no golang stacktrace.. it did happen right after a `cache_flushed_alert` though... might be worth it to make it crash until we do get a stacktrace, but I might need to make it recover less quickly too.

**Comment by scakemyer** [comment]
Closing since the initial issue should be fixed in 0.9.50 :)
